### PR TITLE
All in on pyproject.toml

### LINKED ISSUE #86: Modern installation method?
Related to #81 , the use of distutils is deprecated and will be removed in python 3.12 (https://peps.python.org/pep-0632/), and setup.py files have also been deprecated in favour of pyproject.toml files (https://packaging.python.org/en/latest/tutorials/packaging-projects/, https://peps.python.org/pep-0621/, https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html).

Should we consider moving to a pyproject.toml, and using `pip install -e .` as the recommended install command rather than the deprecated `python setup.py install`?

### PR DESCRIPTION
**Pull request by robertdstein**
I'm not sure about this one. This fixes #86, and unifies all config stuff into a single file (`pyproject.toml`). It also includes e.g settings for linting, so blocks #102. However, versioneer does not work without a `setup.py`. There is setuptools_scm, which I used here instead, that serves the same purpose I THINK (https://github.com/pypa/setuptools_scm). But I would appreciate a second opinion on whether this suitably preserves the desired versioning behaviour.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "gwemopt"
+authors = [
+    {name = "Michael Coughlin", email = "michael.coughlin@ligo.org"},
+]
+description = "A python package for GW-EM Followup Optimization"
+readme = "README.md"
+requires-python = ">=3.10"
```
Comment (mcoughlin) [inline pyproject.toml:12]: Is 3.8 and 3.9 actually not usable or just untested? The standard linux VMs we build in Docker containers will not necessarily have this high a version

**Review by mcoughlin — COMMENTED** [review]
@Theodlz can you weigh in here on the selection of how we take care of the packaging?

File: pyproject.toml (hunk 1 above)
Comment (robertdstein) [reply pyproject.toml:12]: I can check this, I may have used type hints on a few new functions but I can regress them if needed

Comment (mcoughlin) [reply pyproject.toml:12]: Ok that would be great. I am glad we are moving to something more modern. Looks like 3.8 is what is in the SP VM

**Pull request merged**